docs/knowledgebase/runtime/errors.md

```diff
@@ -0,0 +1,66 @@
+---
+title: Runtime Errors
+---
+
+Runtime code should explicitly and gracefully handle all error cases, which is to say that runtime
+code **must** be "non-throwing", or must never
+"[panic](https://doc.rust-lang.org/book/ch09-03-to-panic-or-not-to-panic.html)" to use Rust
+terminology. A common idiom for writing non-throwing Rust code is to write functions that return
+[`Result` types](https://substrate.dev/rustdocs/v2.0.0/frame_support/dispatch/result/enum.Result.html).
+The `Result` enum type possesses an `Err` variant that allows a function to indicate that it failed
+to execute successfully without needing to panic. Dispatchable calls in the FRAME system for runtime
+development _must_ return a
+[`DispatchResult` type](https://substrate.dev/rustdocs/v2.0.0/frame_support/dispatch/type.DispatchResult.html)
+that _should_ be a
+[`DispatchError`](https://substrate.dev/rustdocs/v2.0.0/frame_support/dispatch/enum.DispatchError.html)
+if the dispatchable function encountered an error.
+
+Each FRAME pallet may define custom `DispatchError`s by using
+[the `decl_error!` macro](macros#decl_error).
+
+```rust
+// Errors inform users that something went wrong.
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+		/// Error names should be descriptive.
+		InvalidParameter,
+		/// Errors should have helpful documentation associated with them.
+		OutOfSpace,
+	}
+}
+```
+
+In order to emit custom errors from a pallet, the pallet's module must configure the `Error` type.
+
+```rust
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		// Errors must be initialized if they are used by the pallet.
+		type Error = Error<T>;
+
+		/* snip */
```

Comment:
```suggestion
		/* --snip-- */
```

Reply:
We kinda used a few variants of `snip` in order to be consistent. We are using this options depending on the file type:
```
// --snip--
/* --snip-- */
# --snip--
```